src/app/components/shared/sign-up/sign-up.component.ts

```diff
@@ -45,6 +47,37 @@ export class SignUpComponent {
   }
 
   submit() {
+    if (this.signupForm.valid) {
+      this.userService.getAllUsers().subscribe(res => {
+        for (let i = 0; i < res.length; i++) {
```

Comment:
this method will fail check user exsists. because this for loop will success the very first index is not equal to the entered email.